### PR TITLE
Integrated login and user registration functionality

### PR DESCRIPTION
**Pull request by linajin03**
I have integrated login and user register functionality to WelcomePage and refactored UserRegister classes into a package in UseCases, Controller, GUI, and ExternalInterface to implement a Clean Architecture structure. You should now be able to log in with user "Herman1" and "Password". Though I think username is case sensitive.

### CODE REVIEW

**Comment by linajin03** [comment]
Tests failed because of ItemSearchTests I think.

File: src/main/java/GUI/Listeners/WelcomePageActionListenerNavigation.java [hunk 1]
```diff
@@ -1,6 +1,8 @@
 package GUI.Listeners;
+import GUI.HomePage;
+import UseCases.LoginAction;
+
 import javax.swing.*;
-import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
```
Comment (Ljime) [inline src/main/java/GUI/Listeners/WelcomePageActionListenerNavigation.java:7]: Javadoc for class here

**Review by chens408 — APPROVED** [review]

File: src/main/java/UseCases/UserRegister/UserRegisterCreateUser.java [hunk 2]
```diff
@@ -1,3 +1,5 @@
+package UseCases.UserRegister;
+
 public interface UserRegisterCreateUser {
```
Comment (Ljime) [inline src/main/java/UseCases/UserRegister/UserRegisterCreateUser.java:3]: Java doc here

**Review by Ljime — APPROVED** [review]

**Review by kevinle623 — APPROVED** [review]

File: src/main/java/GUI/ListOfWishlistsPage.java [hunk 3]
```diff
@@ -1,164 +1,164 @@
...
-    }
-
-//    public JButton createButton(String text, int prefWidth, int prefHeight, Color background, Color foreground, Font font) {
-//        JButton button = new JButton();
-//        button.setText(text);
-//        button.setFont(font);
-//        button.setBorder(new GUI.GUI.RoundedBorder(20));
-//        if (prefHeight > 0 && prefWidth > 0) {
-//            button.setPreferredSize(new Dimension(prefWidth, prefHeight));
+//package GUI;
```
Comment (PyroPoro) [inline src/main/java/GUI/ListOfWishlistsPage.java:1]: Please remove commented code if it is not needed

File: src/main/java/GUI/SignupPanel.java [hunk 4]
```diff
@@ -2,15 +2,21 @@
```
Comment (PyroPoro) [inline src/main/java/GUI/SignupPanel.java:2]: java doc pls

File: src/main/java/GUI/WelcomePage.java [hunk 5]
```diff
@@ -1,32 +1,33 @@
 package GUI;
```
Comment (PyroPoro) [inline src/main/java/GUI/WelcomePage.java:1]: java doc

**Review by PyroPoro — APPROVED** [review]

**Pull request merged**